jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,155 +5,430 @@
-    pub policy_stores: HashMap<String, PolicyStore>,
-}
-
-/// PolicyStore contains all the data the Cedarling needs to verify JWT tokens and evaluate policies
+/// Represents the store of policies used for JWT validation and policy evaluation in Cedarling.
+///
+/// The `PolicyStore` contains the schema and a set of policies encoded in base64,
+/// which are parsed during deserialization.
 #[derive(Debug, Clone, serde::Deserialize)]
 pub struct PolicyStore {
-    pub schema: CedarSchema,
-    #[serde(deserialize_with = "deserialize_policy_set::parse")]
-    pub policies: cedar_policy::PolicySet,
+    /// The cedar version to use when parsing the schema and policies.
+    #[serde(deserialize_with = "check_cedar_version")]
+    #[allow(dead_code)]
+    pub cedar_version: String,
+
+    /// Cedar schema in base64-encoded string format.
+    ///
+    /// Extracted from the `policy_store.json` file.
+    pub cedar_schema: CedarSchema,
+
+    /// Cedar policy set in base64-encoded string format.
+    ///
+    /// Extracted from the `policy_store.json` file and deserialized using `deserialize_policies`.
+    #[serde(deserialize_with = "parse_cedar_policy")]
+    pub cedar_policies: cedar_policy::PolicySet,
+
+    /// An optional list of trusted issuers.
+    ///
+    /// This field may contain issuers that are trusted to provide tokens, allowing for additional
+    /// verification and security when handling JWTs.
+    #[allow(dead_code)]
+    pub trusted_issuers: Option<Vec<TrustedIssuer>>,
 }
 
-// Deserialization to [`cedar_policy::PolicySet`] moved here
-// to be close to structure where it is used.
-// And current module is small so it is OK to store it here.
-mod deserialize_policy_set {
+/// Represents a trusted issuer that can provide JWTs.
+///
+/// This struct includes the issuer's name, description, and the OpenID configuration endpoint
+/// for discovering issuer-related information.
+#[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)]
+pub struct TrustedIssuer {
+    /// The name of the trusted issuer.
+    pub name: String,
+
+    /// A brief description of the trusted issuer.
+    pub description: String,
+
+    /// The OpenID configuration endpoint for the issuer.
+    ///
+    /// This endpoint is used to obtain information about the issuer's capabilities.
+    pub openid_configuration_endpoint: String,
+
+    /// Optional metadata related to the tokens issued by this issuer.
+    ///
+    /// This field may include role mappings that help define the access levels for issued tokens.
+    #[serde(deserialize_with = "check_token_metadata")]
+    pub token_metadata: Option<Vec<TokenMetadata>>,
+}
 
-    use std::collections::HashMap;
+/// Validates the `token_metadata` field.
+///
+/// This function ensures that the metadata contains at most one `TokenMetadata` with a `role_mapping`
+/// to prevent ambiguous role assignments.
+fn check_token_metadata<'de, D>(deserializer: D) -> Result<Option<Vec<TokenMetadata>>, D::Error>
```

Comment:
Minor-ish naming quibble: this functions parses and then checks. Name implies that it checks only.

Reply:
renamed the function to `parse_and_check_token_metadata` in [df336b1](https://github.com/JanssenProject/jans/pull/9910/commits/df336b1aaa69a8a8a47527f02d7453d89a7ddc4b):
```rust
/// Parses and validates the `token_metadata` field.
///
/// This function ensures that the metadata contains at most one `TokenMetadata` with a `role_mapping`
/// to prevent ambiguous role assignments.
fn parse_and_check_token_metadata<'de, D>(
    deserializer: D,
) -> Result<Option<Vec<TokenMetadata>>, D::Error>
where
    D: Deserializer<'de>,
```